### PR TITLE
Fixed several bugs, created a dummy code snippet

### PR DESCRIPTION
**Pull request by SermanVS**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,6 +52,41 @@ You can download the sources, build them using `wheel` and install a local packa
...
+```python
+from llamba.chat_model import BaseModel
+from llamba.bioage_model import BioAgeModel
+from llamba.connector import Connector
+import torch
+from torch import nn
+import pandas as pd
+
+# Prepare a BioAge model
+class SampleBioAgeModel(nn.Module): pass
```
Comment (mike-live) [inline README.md]: Вот тут не очевидно как пустая модель работает и что делает =)

File: README.md [hunk 2]
```diff
@@ -52,6 +52,41 @@ You can download the sources, build them using `wheel` and install a local packa
...
+
+model = SampleBioAgeModel()
+bioage_model = BioAgeModel(model)
+
+# Prepare a Chatbot model
+chat_model = BaseModel()
+
+# Initialize a connector
+connector = Connector(bioage_model=bioage_model, chat_model=chat_model)
```
Comment (mike-live) [inline README.md]: Надо какое-то другое слово вместо connector/Connector. А то как-то из кода не ассоциируется с нужным действием. То есть слишком общее название, а в реальности оно очень узкое понятие обозначает

File: README.md [hunk 3]
```diff
@@ -52,6 +52,41 @@ You can download the sources, build them using `wheel` and install a local packa
...
+
+# Prepare data to analyze
+features = torch.rand(10, dtype=torch.float32)
```
Comment (mike-live) [inline README.md]: Тут я бы 10 вынес в переменную, плюс ниже ещё есть она. А 10 это рандомное число или оно чем-то обусловлено?

File: samples/dummy.ipynb [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "h:\\Lobachevsky\\llamba\\llamba_env\\Lib\\site-packages\\tqdm\\auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
```
Comment (mike-live) [inline samples/dummy.ipynb]: Этого тут не надо

File: samples/dummy.ipynb [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    "\n",
+    "data = pd.DataFrame([{f'Feature_{i}' : features[i] for i in range(10)}])\n",
+    "data['Age'] = age\n",
+    "\n",
+    "connector.analyze(data)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
```
Comment (mike-live) [inline samples/dummy.ipynb:66]: Надо его запустить и показать что в выводе

File: requirements.txt [hunk 6]
```diff
@@ -6,6 +6,7 @@ scikit-learn
 seaborn
 torch
 pytorch-lightning
+openpyxl
```
Comment (mike-live) [inline requirements.txt:9]: Вот это я не очень понял

File: llamba/connector.py [hunk 7]
```diff
@@ -16,8 +16,6 @@ def analyze(self, data: pd.DataFrame, **kwargs):
         answer = ''
         data['bio_age'] = self.bioage_model.inference(data.drop(['Age'], axis=1).T)
         acceleration = data['bio_age'].values - data['Age'].values
-        print(data)
-        print(acceleration)
         answer += 'You biological age is {age} and your aging acceleration is {acceleration}, which means '.format(age=round(data['bio_age'].values[0]), acceleration=round(acceleration[0]))
```
Comment (mike-live) [inline llamba/connector.py]: Лучше везде использовать bio_age или слитно bioage. Эта строка очень длинная и просится разбиться чтобы быть не длиннее 90 символов. И заодно f string

**Review by mike-live — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (SermanVS) [reply README.md]: Так она пустышка специально...

File: requirements.txt (hunk 6 above)
Comment (SermanVS) [reply requirements.txt:9]: Для семпла нужно. Там данные считываем из txai_omics_3, без него pandas не хочет работать

File: llamba/connector.py (hunk 7 above)
Comment (SermanVS) [reply llamba/connector.py]: Тут f-string некрасиво будет, там имена параметров слишком длинные, сама строчка нечитаемой станет, но разобью чтоб короче было

File: samples/dummy.ipynb (hunk 5 above)
Comment (SermanVS) [reply samples/dummy.ipynb:66]: У пустышки пустой вывод, hence the name

File: README.md (hunk 3 above)
Comment (SermanVS) [reply README.md]: Это рандомное число, которое ни на что не влияет. Просто пример типа "вот у нас условно 10 признаков"

File: README.md (hunk 2 above)
Comment (SermanVS) [reply README.md]: Я не знаю таких слов, которые могли бы описать четко смысл, при этом не превратившись в предложение

File: llamba/connector.py [hunk 8]
```diff
@@ -1,24 +1,25 @@
...
+    def analyze(self, data: pd.DataFrame, device=torch.device('cpu'), **kwargs):
         answer = ''
-        data['bio_age'] = self.bioage_model.inference(data.drop(['Age'], axis=1).T)
+        data['bio_age'] = self.bioage_model.inference(data=data.drop(['Age'], axis=1), device=device)
         acceleration = data['bio_age'].values - data['Age'].values
-        print(data)
-        print(acceleration)
-        answer += 'You biological age is {age} and your aging acceleration is {acceleration}, which means '.format(age=round(data['bio_age'].values[0]), acceleration=round(acceleration[0]))
+        answer += 'Your bioage age is {age} and your aging acceleration is {acceleration}, which means ' \
+            .format(age=round(data['bio_age'].values[0]),
```
Comment (mike-live) [inline llamba/connector.py]: Напиши не age, а bioage. А то путает

File: llamba/connector.py [hunk 9]
```diff
@@ -1,24 +1,25 @@
 from sklearn.metrics import mean_absolute_error
 import pandas as pd
 import shap
+import torch
 import numpy as np
 
 from llamba.chat_model import BaseModel
```
Comment (mike-live) [inline llamba/connector.py]: А почему она base model? Может ChatModel?

File: llamba/connector.py [hunk 10]
```diff
@@ -1,24 +1,25 @@
...
+class LlambaConnector:
     def __init__(self, bioage_model: BioAgeModel, chat_model: BaseModel):
         self.bioage_model = bioage_model
         self.chat_model = chat_model
 
-    def analyze(self, data: pd.DataFrame, **kwargs):
+    def analyze(self, data: pd.DataFrame, device=torch.device('cpu'), **kwargs):
         answer = ''
-        data['bio_age'] = self.bioage_model.inference(data.drop(['Age'], axis=1).T)
+        data['bio_age'] = self.bioage_model.inference(data=data.drop(['Age'], axis=1), device=device)
```
Comment (mike-live) [inline llamba/connector.py:18]: Транспонирование пропало, надеюсь это ок

File: llamba/bioage_model.py [hunk 11]
```diff
@@ -1,14 +1,16 @@
...
 import torch
+from torch import nn
 
 class BioAgeModel:
-    def __init__(self, model: any):
+    def __init__(self, model: nn.Module):
         self.model = model
 
-    def inference(self, data: pd.DataFrame):
+    def inference(self, device: torch.device, data: pd.DataFrame):
```
Comment (mike-live) [inline llamba/bioage_model.py]: Перестань параметры местами

File: llamba/bioage_model.py [hunk 12]
```diff
@@ -1,14 +1,16 @@
...
+        self.model.to(device)
         self.model.eval()
-        try:
+        if str(device) == "cuda":
+            res = self.model(torch.from_numpy(data.values)).cuda().detach().numpy().ravel()
```
Comment (mike-live) [inline llamba/bioage_model.py:13]: Скорее всего это не будет работать. Надо тестить. Сейчас input не на cuda и модель не на cuda. Возможно что-то ещё не на ней. В общем это не так изи обычно

**Review by mike-live — CHANGES_REQUESTED** [review]
Предлагаю ещё в dummy.ipynb сверху сделать ячейку markdown с описанием того что тут вообще происходит.

File: llamba/connector.py (hunk 10 above)
Comment (SermanVS) [reply llamba/connector.py:18]: Да, оно ломало все

File: llamba/bioage_model.py (hunk 12 above)
Comment (SermanVS) [reply llamba/bioage_model.py:13]: Модель парой строк выше закидывается на девайс.
Ну с данными потом потещу
